[Brian] ip

kubrian opened the pull request:
# Luna

> Here's a fancy inspirational quote

Luna frees your mind for tracking your tasks, supporting:
1. Generic task to be done
1. Deadline for tasks that must be done before a certain time
    * e.g. Writing a _pull request_ for this ip
1. Events which start and end between two timings
    * e.g. **Final** for CS2103 😢

Long term storage is supported via local data files, ensuring that tasks are present even after closing the application.

- [ ] This is an ~~uncompleted~~ task with inline `code` formatting
- [x] This is a completed task with [a link to google](https://www.google.com)

To use the app, first download the jar file, and run it using command line

```shell
cd <path/to/jar>
java -jar luna.jar
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

kubrian committed:
Rename chatbot from Duke to Luna

kubrian committed:
Add greeting

kubrian committed:
Add support to echo user input

kubrian committed:
Add support for storing list and format output

kubrian committed:
Refactor list of tasks to Task.java

kubrian committed:
Task : Add support to mark tasks as done

kubrian committed:
Luna : Refactor cases for input

kubrian committed:
Support three types of tasks

kubrian committed:
Luna : Refactor user interaction and format strings.

kubrian committed:
Luna : Add support to handle invalid user input

kubrian committed:
Luna : Add support to delete task

kubrian committed:
Luna : Add Command enum to specify list of operations

kubrian committed:
Luna : Change access modifiers to internal variables and methods

kubrian committed:
Add basic helpers for saving and loading tasks

kubrian committed:
Bugfix : Remove error when using list/help

kubrian committed:
Add support for saving and loading task list

kubrian committed:
Merge branch 'branch-Level-7'

kubrian committed:
Luna : Refactor imports, variables and for loops

kubrian committed:
Add datetime support for Deadline and Event

kubrian committed:
Merge branch 'branch-Level-8'

kubrian committed:
UI : Add basic support before refactoring

kubrian committed:
Luna : Refactor to use UI

kubrian committed:
Rename UI to ConsoleUI

kubrian committed:
Refactor common messages to Luna

kubrian committed:
Reformat, rename Command, add Storage and Parser

kubrian committed:
Refactor Bye, Help and List as Commands

kubrian committed:
Refactor Todo and remove unused list tasks

kubrian committed:
Refactor Mark, Unmark and Delete as Commands

kubrian committed:
Refactor Deadline, Event as Command

kubrian committed:
Refactor input parsing to Parser

kubrian committed:
Refactor LocalDateTime parsing into Parser

kubrian committed:
Refactor loading and writing file into Storage

kubrian committed:
Merge branch 'branch-moreOOP'

kubrian committed:
Refactor into package

kubrian committed:
Merge branch 'branch-Packages'

kubrian committed:
Merge branch 'add-gradle-support'

kubrian committed:
Update names for Gradle

kubrian committed:
Remove unused functions and update paths for runtest.sh

kubrian committed:
Add tests for Task

kubrian committed:
Rename to ConsoleUi

kubrian committed:
Rename to Todo

kubrian committed:
Remove unused String parsing constructors

kubrian committed:
Check mark completion in tests

kubrian committed:
Add tests for Commands

kubrian committed:
Add test for Parser

kubrian committed:
Merge branch 'junit'

kubrian committed:
Add Javadoc for Commands

kubrian committed:
Add Javadoc

kubrian committed:
Refactor error handling of task saving into Storage

kubrian committed:
Add support for find

kubrian committed:
Merge branch 'javadoc'

kubrian committed:
Add support for find

kubrian committed:
Merge branch 'branch-Level-9'

kubrian committed:
Adhere to coding style

kubrian committed:
Merge branch 'branch-checkstyle'

FabianHeng commented on src/main/java/luna/command/ByeCommand.java:
```diff
@@ -0,0 +1,23 @@
+public class ByeCommand implements Command {
+
+    /**
+     * Executes the command to exit the application.
```
Perhaps you can add the @param and @return tags in the JavaDocs?

kubrian replied:
Not quite sure what's the best practice, but I deliberately did not include this as tags can be automatically inherited and I added both tags in Command#execute.

FabianHeng commented on src/main/java/luna/command/EventCommand.java:
```diff
@@ -0,0 +1,45 @@
+     *
+     * @param description The description of the event.
+     * @param from        The start time of the event.
+     * @param to          The end time of the event.
```
I like how you properly defined all tags and explained the function properly.

FabianHeng commented on src/main/java/luna/command/FindCommand.java:
```diff
@@ -0,0 +1,37 @@
+    public FindCommand(String keyword) {
+        this.keyword = keyword;
+    }
+
```
Perhaps you can add JavaDoc comment to explain this function.

FabianHeng commented on src/main/java/luna/command/MarkCommand.java:
```diff
@@ -0,0 +1,37 @@
+     */
+    @Override
+    public boolean execute(ConsoleUi consoleUi, Storage storage, ArrayList<Task> taskList) {
+        taskList.get(taskNumber - 1)
```
Perhaps you can put line 31 with the same line as line 30.

FabianHeng commented on src/main/java/luna/command/UnmarkCommand.java:
```diff
@@ -0,0 +1,37 @@
+    }
+
+    /**
+     * Executes the command to mark a task as not completed.
```
Perhaps you can add the @param and @return tags?

FabianHeng commented on src/main/java/luna/Luna.java:
```diff
@@ -0,0 +1,86 @@
+            }
+
+            try {
+                boolean cont = command.execute(consoleUi, storage, taskList);
```
Perhaps you can rename the boolean variable to a more appropriate one?

kubrian committed:
Add basic GUI support via JavaFX

kubrian committed:
Scaling GUI, add images

kubrian committed:
Add ability for GUI to echo

kubrian committed:
Refactor into fxml

kubrian committed:
Refactor Command to be UI independent and integrate with GUI

kubrian committed:
Merge branch 'branch-Level-10'

kubrian committed:
Add javadoc

kubrian committed:
Use assertions to verify assumptions

Storage, Task List and Console must be not null when utilized by
Commands.

Let's add assert statements before method calls which check that
arguments are not null.

Assertions should be performed as these variables should always be not
null throughout runtime.

kubrian committed:
Improve the precision of Javadoc

Javadoc headers for subclass implementations of Command#execute are not
standardized.

Let's change that by specifying the return value, and apply this
principle to other locations too.

kubrian committed:
Remove cyclic dependency of Storage on Command.

Storage should not depend on Command as Command utilizes Storage to
update the save file.

Let's change the format of the save file, and make the loading of tasks
detect incorrect formats.

kubrian committed:
Reword help functionality

kubrian committed:
Merge pull request #1 from kubrian/branch-A-Assertions

Use assertions to verify assumptions

kubrian committed:
Reword help functionality

kubrian committed:
Merge pull request #2 from kubrian/branch-CodeQuality

Improve code quality

kubrian committed:
Merge pull request #3 from kubrian/branch-extension

Reword help functionality

kubrian committed:
Add workflow for CI

kubrian committed:
Rename command to storage in tests

kubrian committed:
Add declaration of AI usage

kubrian committed:
Fix UI bugs

kubrian committed:
Add UI image

kubrian committed:
Add User Guide

kubrian committed:
Update gradlew

kubrian committed:
Use AM/PM with English locale

kubrian committed:
Specify English locale for Datetime

kubrian committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	src/main/java/luna/task/Task.java
#	src/test/java/luna/task/DeadlineTest.java
#	src/test/java/luna/task/EventTest.java

kubrian committed:
Shift UI.png into the required location

kubrian committed:
Rename boolean variables

kubrian committed:
Merge pull request #5 from kubrian/rename-boolean

[#4] Improve naming of boolean variables

kubrian committed:
Move UI.png
